webapp/src/components/post_type/component.tsx

```diff
@@ -21,163 +19,182 @@ interface Props {
-            flexWrap: 'wrap',
-            rowGap: '8px',
-        },
-        callIcon: {
-            background: '#339970',
-            borderRadius: '4px',
-            padding: '10px',
-            width: '40px',
-            height: '40px',
-        },
-        callEndedIcon: {
-            background: changeOpacity(this.props.theme.centerChannelColor, 0.16),
-            borderRadius: '4px',
-            padding: '10px',
-            width: '40px',
-            height: '40px',
-        },
-        messageWrapper: {
-            display: 'flex',
-            flexDirection: 'column',
-            margin: '0 12px',
-        },
-        message: {
-            fontWeight: 600,
-        },
-        duration: {
-            color: this.props.theme.centerChannelColor,
-        },
-        joinButton: {
-            display: 'flex',
-            alignItems: 'center',
-            color: '#FFFFFF',
-            background: '#339970',
-            borderRadius: '4px',
-            padding: '10px 16px',
-        },
-        leaveButton: {
-            display: 'flex',
-            alignItems: 'center',
-            color: 'rgba(210, 75, 78, 1)',
-            background: 'rgba(210, 75, 78, 0.1)',
-            borderRadius: '4px',
-            padding: '10px 16px',
-        },
-        profiles: {
-            display: 'flex',
-            alignItems: 'center',
-            marginLeft: 'auto',
-        },
-    };
-
-    onJoinCallClick = async () => {
-        if (this.props.connectedID) {
-            this.props.showSwitchCallModal(this.props.post.channel_id);
+const PostType = ({post, connectedID, hasCall, pictures, profiles, showSwitchCallModal}: Props) => {
+    const onJoinCallClick = () => {
+        if (connectedID) {
+            showSwitchCallModal(post.channel_id);
             return;
         }
-        window.postMessage({type: 'connectCall', channelID: this.props.post.channel_id}, window.origin);
-    }
+        window.postMessage({type: 'connectCall', channelID: post.channel_id}, window.origin);
+    };
 
-    onLeaveButtonClick = () => {
+    const onLeaveButtonClick = () => {
         if (window.callsClient) {
             window.callsClient.disconnect();
         }
-    }
-
-    render() {
-        const subMessage = this.props.post.props.end_at ? (
-            <div>
-                <span style={this.style.duration}>
-                    {`Ended at ${moment(this.props.post.props.end_at).format('h:mm A')}`}
-                </span>
-                <span style={{margin: '0 4px'}}>{'•'}</span>
-                <span style={this.style.duration}>
-                    {`Lasted ${moment.duration(this.props.post.props.end_at - this.props.post.props.start_at).humanize(false)}`}
-                </span>
-            </div>
-        ) : (
-            <span style={this.style.duration}>{moment(this.props.post.props.start_at).fromNow()}</span>
-        );
-
-        return (
-            <div
-                className='call-thread'
```

Comment:
Could we keep the class? This is actually used by e2e tests to select the element.

Reply:
Oh! Hah, sorry. Will change that to a `data-testid` then (or I'll probably try to delete it in the future when I can't find the css class it belongs to ;) )

Reply:
I can't seem to get the playwright tests to run (they keep stalling), what is the command you use?

Reply:
That would be `npx playwright test` from within the `e2e` directory. Just bear in mind I was getting some failures earlier on which are unrelated to this PR. I'd suggest to only test the relevant test by temporarily adding `.only`, here.:

https://github.com/mattermost/mattermost-plugin-calls/blob/f1a38f10c9b9523ab3150bb9ddc773b7ac9b6129/e2e/tests/join_call.spec.ts#L54


```js
test.only('call thread', async ({page}) => {  ...
```

Reply:
Also that assumes MM is running at `http://localhost:8065` with the plugin deployed and enabled. You can pass a custom URL with the `MM_SITE_URL=` env var.